.github/workflows/ci-build.yaml

```diff
@@ -156,6 +156,7 @@ jobs:
         run: |
           set -x
           export GOPATH=$(go env GOPATH)
+          git checkout -- go.mod go.sum
```

Comment:
Can you please explain why this change is required?

Reply:
the extra packages installed here pollute the `go.mod`. 
https://github.com/argoproj/argo-cd/pull/3899/files/6af3c879a4ae1d43486f3c65d959d2506dd14f9b#diff-f3fd5fd1f2977ef196a7b2e80a1da3eaR149-R150

This is to reset them to what their original state on the commit.